packages/client/lib/net/protocol/ethprotocol.ts

```diff
@@ -56,8 +56,6 @@ const messages: Message[] = [
     name: 'BlockHeaders',
     code: 0x04,
     encode: (headers: BlockHeader[]) => headers.map((h) => h.raw()),
-    decode: (headers: BlockHeaderBuffer[]) =>
-      headers.map((h) => BlockHeader.fromValuesArray(h, {})),
```

Comment:
Could we just add common here instead of removing the decode function?

`headers.map((h) => BlockHeader.fromValuesArray(h, { common: this.chain._common }))`

This commit (https://github.com/ethereumjs/ethereumjs-monorepo/pull/1088/commits/2ca45cda37865b15e5da650ce85162729314a6b3) feels like it's moving around and adding code to handle the buffers that was otherwise nicely handled here.

Reply:
I had this implemented before (as announced here https://github.com/ethereumjs/ethereumjs-monorepo/issues/1086#issuecomment-774657947) but then reverted it because I had to pass `config` along every single `decode` method due to typing and the add another round of //es-lint-ignore comments since it wasn't used most of the time. So I decided that this would at the end be the cleaner way, especially since this was in an inconsistent state anyhow with the Block decoding also working without a decode method like now done here for the `BlockHeader` decoding. Wouldn't be a big fan of reverting, but if you have got a cleaner way of doing this feel free to push here, no emotional connections to this code 😋 , just would want this to work "somehow".

Reply:
Maybe this can be readded (then also together with the Block decoding) along some refactor at some point towards some more type friendly handling of this whole ETH/LES message call handling?

Reply:
oh yeah passing common into decode would be a lot of passing around. can't we use the common from `chain` in `EthProtocolOptions`?

> Maybe this can be readded (then also together with the Block decoding) along some refactor at some point towards some more type friendly handling of this whole ETH/LES message call handling?

yes for sure, I will start thinking about this. I think if we inline the `messages` const into methods on the actual class then it would allow for seamless type info without needing to define the additional type helper `EthProtocolMethods`.

Reply:
Ah, yes, you are so right, I didn't see the option that `messages` could be relatively easily inlined into the `Protocol` classes. Have removed the old commits via rebase and done this now in 08ffcd8. I had to add a `no-invalid-this` lint ignore rule here since eslint was complaining, but I checked with a `console.log()` in the decode message and `this` is properly resolved and the `common` object is present and not set to `undefined` or something.

This is really much nicer now. 😄